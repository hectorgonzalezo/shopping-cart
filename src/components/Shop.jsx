import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import uniqid from 'uniqid';
import products from '../assets/products.json';

function Shop({ opaque }) {
  const [items, setItems] = useState([]);

  useEffect(() => {
    // update items to be displayed by shop from json file only if items is empty
    if (items.length === 0) {
      setItems(Object.values(products));
    }
  }, []);

  return (
    <main id="shop" className={opaque ? 'opaque' : ''}>
      <h1>this is the shop</h1>
      {items.map((item) => <h1 key={uniqid()}>{item.name}</h1>)}
    </main>
  );
}

Shop.defaultProps = {
  opaque: false,
};

Shop.propTypes = {
  opaque: PropTypes.bool,
};

export default Shop;
