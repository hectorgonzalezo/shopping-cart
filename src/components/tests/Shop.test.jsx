import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import Shop from '../Shop';

describe('Shop component', () => {
  it('Renders correctly', () => {
    render(
      <MemoryRouter>
        <Routes>
          <Route path="/" element={<Shop />} />
        </Routes>
      </MemoryRouter>
    );

    expect(screen.getByRole('main')).toHaveAttribute('id', 'shop');
    expect(screen.container).toMatchSnapshot();
  });

  it('Clicking on a ShopItem takes you to that item display', () => {
    render(
      <MemoryRouter>
        <Routes>
          <Route path="/" element={<Shop />} />
        </Routes>
      </MemoryRouter>
    );

    const aHeading = screen.getByRole('link', { name: 'Teensy 2.0 16 MHz AVR 5V signals' });
    expect(aHeading.href).toBe('http://localhost/1');
    expect(screen.container).toMatchSnapshot();
  });
});
