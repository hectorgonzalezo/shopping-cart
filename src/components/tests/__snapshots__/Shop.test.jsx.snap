// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Renders correctly 1`] = `undefined`;
